samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -435,32 +459,59 @@ private boolean deQueueAction(ContainerPlacementRequestMessage requestMessage) {
+    String errorMessagePrefix = ContainerPlacementMessage.StatusCode.BAD_REQUEST + " reason: ";
     Boolean invalidAction = false;
     String errorMessage = null;
-    if (standbyContainerManager.isPresent()) {
-      errorMessage = String.format("%s not supported for hot standby enabled", errorMessagePrefix);
+    if (!samzaApplicationState.runningProcessors.containsKey(requestMessage.getProcessorId()) &&
+        !samzaApplicationState.pendingProcessors.containsKey(requestMessage.getProcessorId())
+    ) {
+      errorMessage = String.format("%s invalid processor id neither in running or pending processors", errorMessagePrefix);
       invalidAction = true;
     } else if (placementRequestsCache.containsKey(requestMessage.getUuid())) {
       errorMessage = String.format("%s duplicate UUID of the request, please retry", errorMessagePrefix);
       invalidAction = true;
-    } else if (Integer.parseInt(requestMessage.getProcessorId()) >= samzaApplicationState.processorCount.get()
-    ) {
-      errorMessage = String.format("%s invalid processor id", errorMessagePrefix);
+    } else if (standbyContainerManager.isPresent() && !standbyContainerManager.get()
+        .checkStandbyConstraints(requestMessage.getProcessorId(), requestMessage.getDestinationHost())) {
+      errorMessage = String.format("%s destination host does not meet standby constraints", errorMessagePrefix);
       invalidAction = true;
     }
 
     if (invalidAction) {
-      LOG.info(errorMessage);
       return new ImmutablePair<>(ContainerPlacementMessage.StatusCode.BAD_REQUEST, errorMessage);
     }
 
     return new ImmutablePair<>(ContainerPlacementMessage.StatusCode.ACCEPTED, "Request is accepted");
   }
 
+  /**
+   * An active and a standby container cannot have a concurrent placement action in flight, if they do actions are taken
+   * serially in the order of timestamps
+   */
+  private boolean checkStandbyOrActiveContainerHasActivePlacementAction(String processorId) {
+    if (standbyContainerManager.isPresent()) {
+      // If requested placement action is on a standby container and its active container has a placement request,
+      // this request shall not be de-queued until in-flight action on active container is complete
+      if (StandbyTaskUtil.isStandbyContainer(processorId) && hasActiveContainerPlacementAction(
+          StandbyTaskUtil.getActiveContainerId(processorId))) {
+        return true;
+      }
+      // If requested placement action is on a standby container and its active container has a placement request,
+      // this request shall not be de-queued until in-flight action on active container is complete
+      if (!StandbyTaskUtil.isStandbyContainer(processorId)) {
+        for (String standby : standbyContainerManager.get().getStandbyList(processorId)) {
+          if (hasActiveContainerPlacementAction(standby)) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
```

Comment:
This method can be simplified/inlined with hasActiveAction?

Reply:
done